code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -38,7 +38,8 @@
 	. = ..()
 	for(var/datum/mutation/human/HM in dna.mutations)
 		HM.on_move(src, NewLoc)
-
+	if(dna && dna.species)
+		dna.species.on_move(src)

```

Comment:
there is a moved signal for all atoms, register for that, don’t' add your own hook